src/lib/__tests__/ensureSSL.test.ts

```diff
@@ -0,0 +1,31 @@
+import { ensureSSL } from "../ensureSSL"
+
+describe("Ensure SSL middleware", () => {
+  let req
+  let res
+  let next
+
+  beforeEach(() => {
+    req = { params: {}, logout: jest.fn(), url: "/terms" }
+    res = { redirect: jest.fn() }
+    next = jest.fn()
+  })
+
+  it("redirects normal http requests to https", () => {
+    req.protocol = "http:"
+
+    ensureSSL(req, res, next)
+    expect(res.redirect.mock.calls[0]).toEqual([
+      301,
+      "https://foobar.com/terms",
+    ])
+  })
+
+  it("does not redirect https protocols", () => {
+    req.protocol = "https:"
```

Comment:
I think the problem is the line `req.protocol = "https:"` 

From the Express docs https://expressjs.com/en/api.html#req.protocol

> When the trust proxy setting does not evaluate to false, this property will use the value of the X-Forwarded-Proto header field if present. This header can be set by the client or by the proxy.

The app should use the `X-Forwarded-Proto` header and so the trust proxy setting should be truthy.

Also this should be checking for `https` _not_ `https:`  (the protocol does not include the `:` delimiter